src/element/reader.rs

```diff
@@ -94,6 +101,18 @@ impl<'a, R: ElementReader + ?Sized> Iterator for ElementIterator<'a, R> {
     }
 }
 
+pub struct OwnedElementIterator<R: ElementReader> {
+    reader: R,
+}
```

Comment:
🗺️ `Reader` already had a non-owning iterator, but we needed a version that would take ownership of the `ElementReader` implementation to extend its lifetime.